src/html/fetch-markdown.js

```diff
@@ -13,17 +13,58 @@ const client = require('request-promise');
 
 const GH_RAW = 'https://raw.githubusercontent.com/';
 
-function fetch({ request }, { REPO_RAW_ROOT: rootPath = GH_RAW } = {}, logger) {
+function bail(logger, message) {
+  logger.error(message);
+  return { error: { message } };
+}
+
+function uri(rootPath, owner, repo, ref, path) {
+  const cleanroot = rootPath.replace(/\/$/, '');
```

Comment:
I don't like this...to much regexp...potentially slow. I would just concatenate everything and replace `\/+` by '/'.
or use https://www.npmjs.com/package/uri-js to build proper uris